docs/proposal-component-names.md

```diff
@@ -0,0 +1,60 @@
+# Proposal (Feb 2020)
+
+## Cylc 8 System Component Naming
+
+Cylc is now a complex multi-component system. It is important that we settle on
+consistent, unambiguous, and where possible standard, component names so that
+users and developers can communicate without confusion. (This was not easy even
+before Cylc 8!).
+
+## Proposed Final Names
+
+Note we have already agreed to drop **suite** for **workflow** ("suite" is
+non-standard outside of the NWP community).
+
+| Component Name               | Code Repository   | Config File                                          | Description |
+| --                           |----               | ---                                                  |---          |
+|:boom: **Cylc Scheduler**     | cylc-flow         | :boom: `cylc-flow.conf` and `cylc-flow-global.conf`  | an instance of the cylc-flow scheduler program, to manage a single workflow | 
```

Comment:
```diff
- `cylc-flow.conf` and `cylc-flow-global.conf`
+ `flow.rc`
```

Reply:
Re `.rc` - see my arguments against that in the "Explanatory Comments" bit of the document.

Are  you suggesting `flow.[ext]` for BOTH workflow-specific and global config, even though we are no longer unifying the files?  Not sure about that, as we won't be able to tell which is which by merely inspecting the filename (they might not always be stored in the their correct location).

I could go for `flow.conf` (global) and `cylc-flow.conf` (workflow) IF the site-global location includes `cylc` in the directory hierachy like the user-global location does - but I don't think we've talked about that yet?

Reply:
> see my arguments against that in the "Explanatory Comments" bit of the document.

Just read. I think this change is probably all pain, no gain.

Users are going to hate us for renaming `suite.rc` in the first place.

> Are you suggesting flow.[ext] for BOTH workflow-specific and global config

Meh.

> I could go for flow.conf (global) and cylc-flow.conf (workflow) 

At the moment we are using `flow.rc` for both, I don't really mind if we call them the same thing or use different names.

I don't like `cylc-flow.conf` for workflow definitions, the `cylc-` is superfluous and more to type.

Why not `~/.cylc/flow/global.conf` (global) and `flow.conf` (workflow)?

"We refer to workflows as flows which are defined a in a `flow.conf` file"

Reply:
> Users are going to hate us for renaming suite.rc in the first place.

Huh, well, too bad - we can't standardize our terminology for the longer-term good without getting rid of the word "suite", and we agreed to do that long ago. So users already have to change their workflow configuration file names.  So why not make them change the last two  characters *in a filename that they already have to change the rest of* - to a suffix that is also more standard and clearly indicates a config file (no one knows what ".rc" means any more).

Reply:
> I don't like `cylc-flow.conf` for workflow definitions, the `cylc-` is superfluous and more to type.
> Why not ~/.cylc/flow/global.conf (global) and flow.conf (workflow)?

The `cylc-` is only five characters, you don't have to type it very often, and it is only superfluous if the directory hierachy contains "cylc" - which it does for the user global file, but not for the workflow file, and it may or may not for the site global file.
  
Plain `global.conf` is extremely generic - if you find one of those lying around god only knows what it is for. 

For the workflow config `flow.conf` also seems a bit too generic to me, given that half the workflow engines on the planet are likely to call their workflows "flows" too.  Why not make it completely obvious what the file is for with only five extra characters? `cylc-flow.conf`